zls crashes hard on startup with the new semantic token support

With the latest CI build, when opening a .zig file with zls (`sh -c zls`), the server crashes on initialization:

```
[Trace - 10:21:23] Sending request 'initialize - (1)'.
Params: {
  "processId": 6332,
  "rootPath": "/Users/fbricon/Dev/souk/IdeaProjects/lsp4ij-tests/untitled",
  "rootUri": "file:///Users/fbricon/Dev/souk/IdeaProjects/lsp4ij-tests/untitled",
  "initializationOptions": {},
  "capabilities": {
    "workspace": {
      "applyEdit": true,
      "workspaceEdit": {
        "documentChanges": true,
        "resourceOperations": [
          "create",
          "delete",
          "rename"
        ]
      },
      "didChangeConfiguration": {
        "dynamicRegistration": true
      },
      "didChangeWatchedFiles": {
        "dynamicRegistration": true
      },
      "executeCommand": {
        "dynamicRegistration": true
      },
      "workspaceFolders": true,
      "configuration": true,
      "semanticTokens": {
        "refreshSupport": true
      },
      "fileOperations": {
        "didRename": true,
        "willRename": true,
        "dynamicRegistration": true
      },
      "inlayHint": {
        "refreshSupport": true
      }
    },
    "textDocument": {
      "synchronization": {
        "willSave": true,
        "willSaveWaitUntil": true,
        "didSave": true
      },
      "completion": {
        "completionItem": {
          "snippetSupport": true,
          "documentationFormat": [
            "markdown",
            "plaintext"
          ],
          "deprecatedSupport": true,
          "resolveSupport": {
            "properties": [
              "documentation",
              "detail",
              "additionalTextEdits"
            ]
          },
          "insertTextModeSupport": {
            "valueSet": [
              1,
              2
            ]
          },
          "labelDetailsSupport": true
        },
        "completionList": {
          "itemDefaults": [
            "editRange"
          ]
        }
      },
      "hover": {
        "contentFormat": [
          "markdown",
          "plaintext"
        ]
      },
      "signatureHelp": {
        "signatureInformation": {
          "parameterInformation": {
            "labelOffsetSupport": true
          }
        }
      },
      "references": {},
      "documentHighlight": {},
      "formatting": {
        "dynamicRegistration": true
      },
      "rangeFormatting": {
        "dynamicRegistration": true
      },
      "declaration": {
        "linkSupport": true
      },
      "definition": {
        "linkSupport": true
      },
      "typeDefinition": {
        "linkSupport": true
      },
      "implementation": {
        "linkSupport": true
      },
      "codeAction": {
        "codeActionLiteralSupport": {
          "codeActionKind": {
            "valueSet": [
              "quickfix",
              "refactor",
              "refactor.extract",
              "refactor.inline",
              "refactor.rewrite",
              "source",
              "source.organizeImports"
            ]
          }
        },
        "dataSupport": true,
        "resolveSupport": {
          "properties": [
            "edit"
          ]
        },
        "dynamicRegistration": true
      },
      "codeLens": {},
      "documentLink": {},
      "colorProvider": {},
      "rename": {
        "prepareSupport": true
      },
      "publishDiagnostics": {
        "relatedInformation": true,
        "tagSupport": {
          "valueSet": [
            1,
            2
          ]
        },
        "codeDescriptionSupport": true,
        "dataSupport": true
      },
      "foldingRange": {},
      "semanticTokens": {
        "requests": {
          "range": false,
          "full": true
        },
        "tokenTypes": [
          "namespace",
          "type",
          "class",
          "enum",
          "interface",
          "struct",
          "typeParameter",
          "parameter",
          "variable",
          "property",
          "enumMember",
          "event",
          "function",
          "method",
          "macro",
          "keyword",
          "modifier",
          "comment",
          "string",
          "number",
          "regexp",
          "operator",
          "decorator",
          "label"
        ],
        "tokenModifiers": [
          "declaration",
          "definition",
          "readonly",
          "static"
        ],
        "serverCancelSupport": true
      },
      "inlayHint": {}
    },
    "window": {
      "workDoneProgress": true,
      "showMessage": {},
      "showDocument": {
        "support": true
      }
    }
  },
  "clientInfo": {
    "name": "IntelliJ",
    "version": "PyCharm (build PC-241.18034.82)"
  },
  "trace": "off",
  "workspaceFolders": [
    {
      "uri": "file:///Users/fbricon/Dev/souk/IdeaProjects/lsp4ij-tests/untitled",
      "name": "untitled"
    }
  ]
}
[Trace - 10:21:23] Received response 'initialize - (1)' in 21ms.
No result returned.
Error: {
  "code": -32700,
  "message": "ParseError"
}
org.eclipse.lsp4j.jsonrpc.ResponseErrorException: ParseError
  at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleResponse(RemoteEndpoint.java:209)
  at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:193)
  at com.redhat.devtools.lsp4ij.LanguageServerWrapper.lambda$start$1(LanguageServerWrapper.java:260)
  at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)
  at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796)
  at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
  at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
  at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
  at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
  at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
[Trace - 10:21:23] Sending request 'shutdown - (2)'.
Params: null
[Trace - 10:21:23] Received response 'shutdown - (2)' in 0ms.
No result returned.
Error: {
  "code": -32002,
  "message": "ServerNotInitialized"
}
java.lang.Exception: Error while shutdown the language server '970fa682-af8d-479e-af95-4c1952dca6d3'
  at com.redhat.devtools.lsp4ij.LanguageServerWrapper.shutdownLanguageServerInstance(LanguageServerWrapper.java:520)
  at com.redhat.devtools.lsp4ij.LanguageServerWrapper.shutdownAll(LanguageServerWrapper.java:483)
  at com.redhat.devtools.lsp4ij.LanguageServerWrapper.lambda$stop$11(LanguageServerWrapper.java:452)
  at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)
  at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796)
  at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
  at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
  at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
  at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
  at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.util.concurrent.ExecutionException: org.eclipse.lsp4j.jsonrpc.ResponseErrorException: ServerNotInitialized
  at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
  at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2096)
  at com.redhat.devtools.lsp4ij.LanguageServerWrapper.shutdownLanguageServerInstance(LanguageServerWrapper.java:510)
  ... 9 more
Caused by: org.eclipse.lsp4j.jsonrpc.ResponseErrorException: ServerNotInitialized
  at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleResponse(RemoteEndpoint.java:209)
  at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:193)
  at com.redhat.devtools.lsp4ij.LanguageServerWrapper.lambda$start$1(LanguageServerWrapper.java:260)
  ... 7 more
[Trace - 10:21:23] Sending notification 'exit'
Params: null
java.lang.NullPointerException: Cannot invoke "com.redhat.devtools.lsp4ij.server.StreamConnectionProvider.ensureIsAlive()" because "this.lspStreamProvider" is null
  at com.redhat.devtools.lsp4ij.LanguageServerWrapper.lambda$start$0(LanguageServerWrapper.java:243)
  at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
  at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760)
  at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
  at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
  at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
  at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
  at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
[Trace - 10:21:24] Sending request 'initialize - (1)'.
```

It starts fine with LSP4IJ 0.0.2. The main differences are the new semantic token capabilities added to the initialize payload:
<img width="248" alt="Screenshot 2024-07-02 at 10 38 22" src="https://github.com/redhat-developer/lsp4ij/assets/148698/fed2a0b0-615c-4f9d-843b-8757cb5e8401">
<img width="397" alt="Screenshot 2024-07-02 at 10 38 08" src="https://github.com/redhat-developer/lsp4ij/assets/148698/4de2e1c5-6a07-4de1-aa1d-edf5d7fb15da">

--- a/src/main/java/com/redhat/devtools/lsp4ij/internal/ClientCapabilitiesFactory.java
+++ b/src/main/java/com/redhat/devtools/lsp4ij/internal/ClientCapabilitiesFactory.java
@@ -252,6 +252,7 @@ public class ClientCapabilitiesFactory {
         ));
         semanticTokensCapabilities.setServerCancelSupport(Boolean.TRUE);
         var semanticTokensClientCapabilitiesRequests = new SemanticTokensClientCapabilitiesRequests(Boolean.TRUE, Boolean.FALSE);
+        semanticTokensCapabilities.setFormats(List.of(TokenFormat.Relative));
         semanticTokensCapabilities.setRequests(semanticTokensClientCapabilitiesRequests);
         textDocumentClientCapabilities.setSemanticTokens(semanticTokensCapabilities);
 

